metalearners/drlearner.py

```diff
@@ -331,6 +331,35 @@ def evaluate(
 
         return variant_outcome_evaluation | propensity_evaluation | treatment_evaluation
 
+    def treatment_effect(
+        self,
+        X: Matrix,
+        y: Vector,
+        w: Vector,
+    ) -> np.ndarray:
+        """Compute Average Treatment Effect (ATE) for each treatment variant using the Augmented IPW estimator (Robins et al 1994). Does not require fitting a second-stage treatment model: it uses the pseudo-outcome alone and computes the average and SE. Can be used following the `fit_all_nuisance` method.
+
+        Args:
+            X (Matrix): Covariate matrix.
+            y (Vector): Outcome vector.
+            w (Vector): Treatment vector
+
+        Returns:
+            np.ndarray: Treatment effect and standard error for each treatment variant.
+        """
+        gamma_matrix = np.zeros((len(X), self.n_variants - 1))
+        for treatment_variant in range(1, self.n_variants):
+            gamma_matrix[:, treatment_variant - 1] = self._pseudo_outcome(
+                X=X,
+                w=w,
+                y=y,
+                treatment_variant=treatment_variant,
+                is_oos=False,
+            )
+        treatment_effect = gamma_matrix.mean(axis=0)
+        standard_error = gamma_matrix.std(axis=0) / np.sqrt(len(X))
+        return np.c_[treatment_effect, standard_error]
```

Comment:
Your implementation makes a great impression on me and sure looks slick. Would you mind adding a small test for this functionality to https://github.com/Quantco/metalearners/blob/main/tests/test_drlearner.py?

We could compare the returned values against some simple mock data.

Moreover, I think it could be very useful to make sure everything works as expected:
- with more than two variants
- with classification outcomes

If one of those cases doesn't work with the current implementation, it'd be good if we explicitly checked for that and added it to the doc string.